lib/payment_processor/braintree/webhook_handler.rb

```diff
@@ -51,8 +51,10 @@ def process_notification
         when 'subscription_charged_unsuccessfully'
           handle_failed_subscription_charge
         else
-          Rails.logger.info("Unsupported Braintree::WebhookNotification received of type '#{@notification.kind}'")
+          Rails.logger.info("Unsupported Braintree::WebhookNotification received of type '#{notification.kind}'")
         end
+      rescue
+        Rails.logger.error("Braintree webhook handling failed for '#{notification.kind}', for subscription ID '#{notification.subscription.id}'")
```

Comment:
I'd like to keep this in, since this is basically the information we need from any errors in any of the handling methods and just doing it once in `process_notification` makes us not repeat it all over the place. There is additional logging in the methods called by the individual handling methods (e.g. `customer`) for more information.